CMakeLists.txt

```diff
@@ -26,6 +26,21 @@ option(USE_TSAN "Enable ThreadSanitizer checks" OFF)
 option(USE_MSAN "Enable MemorySanitizer checks" OFF)
 option(USE_VALGRIND "Enable Valgrind instrumentation" OFF)
 
+set(buildTypes Release Debug RelWithDebInfo MinSizeRel)
+
+if(NOT CMAKE_BUILD_TYPE)
+    message(STATUS "No build type selected (CMAKE_BUILD_TYPE), defaulting to Release")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+            "Choose the type of build, options are: Release Debug RelWithDebInfo MinSizeRel ..." FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS ${buildTypes})
+else()
+    message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
+    if(NOT CMAKE_BUILD_TYPE IN_LIST buildTypes)
+        message(WARNING "Unusual build type was set, please make sure it's a proper one. "
```

Comment:
please error here

Reply:
I don't believe it's an error. Someone may want to use custom build type in their project and fetch our project... so it should not fail, just warn

Reply:
Done

Reply:
@ldorau, pls see my comment above. You can verify what Internet says about it

Reply:
OK, reverted.
Done

Reply:
@lukaszstolarczuk OK, reverted.
Done